### PR TITLE
FLOE-497: update to Infusion 2.0

### PR DESCRIPTION
**Pull request by waharnum**
JIRA here: https://issues.fluidproject.org/browse/FLOE-497

This is a basic maintenance update to use the Infusion 2.0 release from NPM and the dev branch of Flocking that's Infusion 2.0-compatible.

### CODE REVIEW

**Comment by waharnum** [comment]
I've done the following:

1. Merged @colinbdclark's PR (https://github.com/fluid-project/chartAuthoring/pull/18) into this one 
2. fixed all the paths to infusion-all.js that @cindyli correctly highlighted as an issue (I believe both @colinbdclark and myself had old files hanging around, which is why it didn't fail for us).
3. per comment 2, added `grunt-contrib-clean` and a `clean:frontEndDependencies` task; this gets run as part of the `installFrontEnd` task and wipes the `lib` directories, so that old files don't hang around and fool us into thinking something is working when it will be broken from a fresh install.
4. Added a section to the README on using testem to run the tests (thanks, @colinbdclark, good change!)
5. Upgraded to the version of Infusion requested by @amb26 and verified that all tests continue to pass and the demo behaves as expected.

I think this can now be reviewed again for merge by @cindyli and/or @amb26.

File: README.md [hunk 1]
```diff
@@ -25,3 +25,11 @@ Steps:
 
 Result:
 * The root directory can now serve up the demos and tests as a static site
+
+##Testing
+
+If you have `testem` installed, you can run the tests as follows:
```
Comment (cindyli) [inline README.md]: Would be helpful to add a link that points to how to install testem.

File: Gruntfile.js [hunk 2]
```diff
@@ -33,21 +33,18 @@ module.exports = function (grunt) {
                     {expand: true, cwd: "./node_modules/d3/", src: "**", dest: "./src/lib/d3/"},
                     // Flocking
                     {expand: true, cwd: "./node_modules/flocking/", src: "**", dest: "./src/lib/flocking/"},
-                    // Infusion
-                    {expand: true, cwd: "./node_modules/infusion/build", src: "**", dest: "./src/lib/infusion"},
+                    // Infusion distribution files
+                    {expand: true, cwd: "./node_modules/infusion/dist", src: "**", dest: "./src/lib/infusion/dist"},
```
Comment (cindyli) [inline Gruntfile.js]: I wonder if it's necessary to copy entire `node_modules/infusion/dist` directory even though `infusion-all.js` is only file required by this project from that directory.

File: README.md [hunk 3]
```diff
@@ -25,3 +25,11 @@ Steps:
...
+
+```
+testem ci --file tests/testem.json
```
Comment (cindyli) [inline README.md]: Since a node script is defined for this command in `package.json`, shall this command be `npm test`?

**Comment by waharnum** [comment]
@cindyli thanks for review - agree on all points & have made revisions accordingly.

**Comment by cindyli** [comment]
Merged at 23232ccb4078f05725879ef030366f280cd90b3e

**Pull request merged**